src/util/coding-jams/managePolls.ts

```diff
@@ -0,0 +1,290 @@
+		return;
+	}
+
+	const poll = pollDb.get(pollKey)!;
+
+	await interaction.reply({ embeds: [pollEmbed(poll, pollKey, "(view)")], ephemeral: true });
+}
+
+export async function deletePoll(interaction: CommandInteraction, name: string) {
+	const pollKey = pollDb.findKey((poll) => poll.title === name);
+
+	if (!pollKey) {
+		interaction.reply({ content: "There is no poll with that name...", ephemeral: true });
+		return;
+	}
+
+	pollDb.delete(pollKey);
+
+	for (const key of pollEventsDb.keyArray()) {
+		const poll = pollEventsDb.get(key)!;
+		if (!(poll.pollID === pollKey)) continue;
+		pollEventsDb.delete(key);
+	}
+
+	interaction.reply({ content: `${name} deleted!`, ephemeral: true });
+}
+
+export async function votesPoll(interaction: CommandInteraction, name: string) {
+	const pollKey = pollDb.findKey((poll) => poll.title === name);
+
+	if (!pollKey) {
+		interaction.reply({ content: "There is no poll with that name...", ephemeral: true });
+		return;
+	}
+	const poll = pollDb.get(pollKey)!;
+
+	if (poll.start > DateTime.now()) {
+		await interaction.reply({ content: "Voting hasn't started yet...", ephemeral: true });
+		return;
+	}
+
+	const votesPerUser = Array.from(poll.votes.entries()).map((e) => ({ user: e[0], votes: e[1] }));
+	const votesPerProposal: { key: string; votes: number }[] = [];
+	const votesInTotal = votesPerUser.map((e) => e.votes).flat();
+
+	poll.proposals.forEach((p) =>
+		votesPerProposal.push({ key: p, votes: votesInTotal.filter((v) => v === p).length })
+	);
+	votesPerProposal.sort((a, b) => b.votes - a.votes);
+
+	const proposals = getFromEnmap(
+		proposalDb,
+		votesPerProposal.map((p) => p.key)
+	);
+	const usersPerProposal = votesPerProposal.map((e) =>
+		votesPerUser.filter((v) => v.votes.includes(e.key)).map((v) => v.user)
+	);
+
+	const list = numberedList(
+		proposals.map((p) => p?.title || "Unknown"),
+		votesPerProposal.map((p) => p.votes + " votes")
+	);
+
+	const embed = new EmbedBuilder().setTitle(`${poll.title} (votes)`).addFields(
+		list.map((e, i) => ({
+			name: e ?? "Unknown",
+			value: usersPerProposal[i].map((u) => `<@${u}>`).join(" ") || "-",
+		}))
+	);
+
+	await interaction.reply({ embeds: [addEmbedFooter(embed)], ephemeral: true });
+}
+
+export function sortBySelectionType(selectionType: string) {
+	const sorting: {
+		[key: string]: (a: { key: string; val: Proposal }, b: { key: string; val: Proposal }) => number;
+	} = {
+		[pollSelectionTypes.random]: () => (Math.random() > 0.5 ? 1 : -1),
+		[pollSelectionTypes.newest]: (a, b) => (a.val.created >= b.val.created ? 1 : -1),
+		[pollSelectionTypes.oldest]: (a, b) => (a.val.created <= b.val.created ? 1 : -1),
+		[pollSelectionTypes.top]: (a, b) => b.val.votesLastPoll - a.val.votesLastPoll,
+		[pollSelectionTypes.bottom]: (a, b) => a.val.votesLastPoll - b.val.votesLastPoll,
+		[pollSelectionTypes.topAll]: (a, b) => b.val.totalVotes - a.val.totalVotes,
+		[pollSelectionTypes.bottomAll]: (a, b) => a.val.totalVotes - b.val.totalVotes,
```

Comment:
Add a filter to place never used proposals to the fron

Reply:
Thats just bottomAll

Reply:
Oh wait no I could also sort by number of polls. Yeah okay